torch/fx/OVERVIEW.md

```diff
@@ -0,0 +1,121 @@
+
+We believe building a component that occupies this point in the space will serve the transformation and lowering customer segments well.
+
+## Use Cases ##
+
+FX should be used by pass writers to provide functionality for capturing and constructing nn.Module code in a structured way. We do not expect end users to utilize FX directly. A useful property of framing FX in this way is that passes can be seen as functions of the form `pass(in_mod : nn.Module) -> nn.Module`. This means we can create composable pipelines of transformations.
+
+![An image of a sample nn.Module transformation pipeline that starts with a Quantize transformation, which is then composed with a Split transformation, then a Lower to Accelerator transformation](https://scontent.xx.fbcdn.net/v/t1.0-9/p720x720/118780346_998065427281509_1321963618820011983_o.png?_nc_cat=104&ccb=2&_nc_sid=32a93c&_nc_ohc=8YbXsAHKRjoAX-Hwj4g&_nc_ht=scontent.xx&_nc_tp=30&uss=d63649d5ad92822b&odm=ZmIud29ya3BsYWNlLmNvbQ&_nc_rmd=260&_nc_log=1&oe2=601C8940&oh=225eeda6b37d2198ffbe409a2276b43d&oe=5FF78064 "nn.Module transformation pipeline")
+
+In this example pipeline, we have a Quantize transformation, which is then composed with a Split transformation, then a Lower to Accelerator transformation. Finally, the transformed Modules are compiled with TorchScript for deployment. This last point emphasizes that not only should FX transforms be composable with each other, but their products are composable with other systems like TorchScript compilation or tracing.
+
+# Symbolic Tracing
+
+## About ##
+
+FX’s front-end makes use of the dynamic nature of Python to intercept call-sites for various entities (PyTorch operators, Module invocations, and Tensor method invocations). This functionality is exposed through an API called `torch.fx.symbolic_trace`.  We can see how this works by way of an example:
+
+```python
+import torch
+
+class MyModule(torch.nn.Module):
+  def __init__(self):
+    super().__init__()
+    self.param = torch.nn.Parameter(
+        torch.rand(3, 4))
+    self.linear = torch.nn.Linear(4, 5)
+
+  def forward(self, x):
+    return self.linear(x + self.param).clamp(min=0.0, max=1.0)
+
+from torch.fx import symbolic_trace
+module = MyModule()
+symbolic_traced : torch.fx.GraphModule = symbolic_trace(module)
+
+input = torch.rand(3, 4)
+torch.testing.assert_allclose(symbolic_traced(input), module(input))
+```
+
+Here, we set up a simple Module that exercises different language features: fetching a parameter, applying an arithmetic operator, applying a submodule (linear), and applying a Tensor method. `symbolic_trace` returns an instance of GraphModule, which is in itself a subclass of `nn.Module`. We can see that the `symbolic_traced` instance runs and returns the same result as the original module instance module.
+
+## Tracer ##
```

Comment:
For each of these sections (`Tracer`, `Proxy`, IR (`Graph`), it's probably a good idea to add anchor links to the respective sections in the API docs (https://pytorch.org/docs/master/fx.html) for further reference